lib/liquid/template.rb

```diff
@@ -64,11 +62,11 @@ class << self
       attr_writer :taint_mode
 
       def file_system
-        @@file_system
+        Thread.current[:liquid_file_system] ||= BlankFileSystem.new
```

Comment:
Oh, indeed, I see the issue now. About overriding the filesystem on instance level -  it's already possible, here: https://github.com/Shopify/liquid/blob/d8bda2c8923c6b3048678016dae49a4fecd963fc/lib/liquid/tags/include.rb#L105

But I'm wondering actually if that isn't a desired behaviour in such case. I think it's less surprising that each thread would require separate assignment of file system or it would default to the default one otherwise than the situation when the threads override the global file system at the same time. Indeed it's a breaking change, but looks like a step in the right direction.